libgadget/density.c

```diff
@@ -317,10 +317,10 @@ density_ngbiter(
         iter->base.symmetric = NGB_TREEFIND_ASYMMETRIC;
         return;
     }
-    int other = iter->base.other;
-    double r = iter->base.r;
-    double r2 = iter->base.r2;
-    double * dist = iter->base.dist;
+    const int other = iter->base.other;
```

Comment:
I suspect this is where most of the gain is from. 

A comment, so these consts are not removed in the future by accident.

Reply:
Actually, no, most of the gain is in the treewalk and shows up in the short-range gravity walk (curiously).

Reply:
Likely because the compiler could figure out these are consts, but couldn't by itself in the other case.

Reply:
That's probably true